mlc_llm/relax_model/chatglm.py

```diff
@@ -0,0 +1,834 @@
+        bb.emit_func_output(gv)
+
+
+def create_softmax_func(bb: relax.BlockBuilder, config: ChatGLMConfig) -> None:
+    with bb.function("softmax_with_temperature"):
+        logits = nn.Placeholder(
+            (1, 1, config.padded_vocab_size), dtype="float32", name="logits"
+        )
+        temperature = nn.Placeholder((), dtype="float32", name="temperature")
+        with bb.dataflow():
+            div = bb.emit(relax.op.divide(logits, temperature))
+            softmax = bb.emit(relax.op.nn.softmax(div, axis=-1))
+            gv = bb.emit_output(softmax)
+        bb.emit_func_output(gv, [logits, temperature])
+
+
+def get_model(args: argparse.Namespace, hf_config):
+    model = args.model
+    dtype = args.quantization.model_dtype
+
+    if model.startswith("chatglm2") or model.startswith("codegeex2"):
+        config = ChatGLMConfig(
+            **hf_config,
+            dtype=dtype,
+        )
+
+        param_manager = ParamManager()
+        bb = relax.BlockBuilder()
+        create_encoding_func(bb, param_manager, config, args.quantization)
+        create_decoding_func(bb, param_manager, config, args.quantization)
+        create_kv_cache_func(bb, config)
+        create_softmax_func(bb, config)
+        create_metadata_func(
+            bb,
+            model_name=model,
+            max_window_size=config.seq_length,
+            stop_tokens=[0],
+            add_prefix_space=False,
+        )
+
+        mod = bb.get()
+        for gv in mod.functions:
+            func = mod[gv]
+            if isinstance(func, relax.Function):
+                mod[gv] = func.with_attr(
+                    "tir_var_upper_bound",
+                    {
+                        "n": config.seq_length,
+                        "m": config.seq_length,
+                    },
+                )
+
+        if args.build_model_only:
+            return mod, param_manager, None
+
+        def f_convert_pname_fwd(pname: str) -> List[str]:
+            if "transformer.embedding" in pname:
+                return [
+                    pname.replace(
+                        "transformer.embedding", "transformer.embedding.word_embeddings"
+                    )
+                ]
+            else:
+                return [pname]
+
+        def f_convert_param_bkwd(torch_pname: str, torch_param):
+            if "transformer.embedding.word_embeddings" in torch_pname:
+                return [
+                    (
+                        torch_pname.replace(
+                            "transformer.embedding.word_embeddings",
+                            "transformer.embedding",
+                        ),
+                        torch_param.astype(dtype),
+                    )
+                ]
+            else:
+                return [(torch_pname, torch_param.astype(dtype))]
+
+        param_manager.set_param_loading_func(
+            args.model_path, args.use_safetensors, f_convert_pname_fwd, f_convert_param_bkwd
```

Comment:
I just needed to provide the safetensors argument to the parameter manager, as it was added in #627. I actually was able to fix it right away, here, in the last commit. Feel free to merge it if looks good!